src/db.c

```diff
@@ -947,6 +946,7 @@ void scanGenericCommand(client *c, robj *o, unsigned long cursor) {
     listNode *node;
     long count = 10;
     sds pat = NULL;
+    sds typename = NULL;
```

Comment:
that var is no longer needed again.

Reply:
I think it is still needed, because we must use it as a flag that if there is a type filter  in redis7.2. And I deleted the local repeated `typename`  just now, so we also no need to delete this line in  redis 8.

Reply:
can't we use `type != LLONG_MAX` like we planed to use for 8.0 (and move that var to the inner scope)?

Reply:
please correct me if i'm wrong, if specificated type filter is unkown in redis7.2, which means `type`  is equal LLONG_MAX, we still need continue to scan the full key space and execute `lookupKey` for these keys. But in redis8.0, we will will reply a error immediately for the unkown type filter, `type != LLONG_MAX` is just means there is no type filter in scan command.

Reply:
ohh, right.
please add a test to cover that.